### ISSUE TITLE
Any item can be placed in armor slots, leading to fun exceptions.

### ISSUE THREAD

**Issue by dequis** [GlowstoneMC/Glowstone-Legacy#105]
![](http://dump.dequis.org/pM8VZ.png)

Those beds in the armor slots are actually the same item, the client gets confused and duplicates them. Some items behave differently - the potatoes don't get duped for example. This is just a visual glitch, reconnecting cleans it up.

```
01:38:41 [INFO] dequis[/127.0.0.1:60682]: WindowClickMessage{id=0, slot=39, button=1, transaction=36, mode=5, item=null} --> false
01:38:41 [SEVERE] Error while handling WindowClickMessage{id=0, slot=-999, button=2, transaction=37, mode=5, item=null} (handler: WindowClickHandler)
java.lang.ArrayIndexOutOfBoundsException: -999
        at net.glowstone.inventory.GlowInventory.getSlotType(GlowInventory.java:99)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.process(WindowClickHandler.java:47)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.handle(WindowClickHandler.java:23)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.handle(WindowClickHandler.java:19)
        at com.flowpowered.networking.session.BasicSession.handleMessage(BasicSession.java:82)
        at com.flowpowered.networking.session.BasicSession.messageReceived(BasicSession.java:141)
        at net.glowstone.net.GlowSession.pulse(GlowSession.java:357)
        at net.glowstone.net.SessionRegistry.pulse(SessionRegistry.java:23)
        at net.glowstone.scheduler.GlowScheduler.pulse(GlowScheduler.java:172)
        at net.glowstone.scheduler.GlowScheduler.access$100(GlowScheduler.java:24)
        at net.glowstone.scheduler.GlowScheduler$2.run(GlowScheduler.java:109)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:304)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:178)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:744)
01:38:41 [INFO] dequis[/127.0.0.1:60682]: WindowClickMessage{id=0, slot=14, button=0, transaction=38, mode=1, item=ItemStack{BEACON x 1}} --> false
01:38:42 [INFO] dequis[/127.0.0.1:60682]: WindowClickMessage{id=0, slot=15, button=0, transaction=39, mode=1, item=ItemStack{DIRT x 1}} --> false
01:38:42 [SEVERE] Error while handling WindowClickMessage{id=0, slot=15, button=0, transaction=40, mode=1, item=null} (handler: WindowClickHandler)
java.lang.NullPointerException
        at net.glowstone.net.handler.play.inv.WindowClickHandler.process(WindowClickHandler.java:186)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.handle(WindowClickHandler.java:23)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.handle(WindowClickHandler.java:19)
        at com.flowpowered.networking.session.BasicSession.handleMessage(BasicSession.java:82)
        at com.flowpowered.networking.session.BasicSession.messageReceived(BasicSession.java:141)
        at net.glowstone.net.GlowSession.pulse(GlowSession.java:357)
        at net.glowstone.net.SessionRegistry.pulse(SessionRegistry.java:23)
        at net.glowstone.scheduler.GlowScheduler.pulse(GlowScheduler.java:172)
        at net.glowstone.scheduler.GlowScheduler.access$100(GlowScheduler.java:24)
        at net.glowstone.scheduler.GlowScheduler$2.run(GlowScheduler.java:109)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:304)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:178)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:744)
```

**Comment by warriordog** [comment]
I found something related:  When shift-clicking items in creative mode they sometimes go to the armor slots, causing these exceptions.

**Comment by SpaceManiac** [comment]
This particular bug was squashed as part of the fix set for #128, as confirmed by SonarBerserk on IRC.
